### ISSUE TITLE
Bayesian optimization of weight initialization

### ISSUE THREAD

**Issue by carlthome** [fchollet/keras#2477]
I think with my current dataset models can converge to a lot of silly solutions so I was thinking one way of finding deeper minima could be train a bunch of models in parallel and iteratively adjust the random weight initialization by Bayesian optimization as a function of validation loss, and restart training. Has anyone attempted something like this before with Keras and perhaps could share an example? Is it worthwhile?

**Comment by kylemcdonald** [comment]
@fchollet mentions in #896 that he's done this hyperparameter optimization in Keras using hyperopt http://hyperopt.github.io/hyperopt/ and there's some example code posted here https://github.com/fchollet/keras/issues/1591#issuecomment-177062588 while others have suggested doing this with https://github.com/HIPS/Spearmint including this issue #2266

I don't know how common it is to optimize the weight initialization scheme vs other hyperparameters, if you check the example code above you can see @jacobzweig is sticking with (what is now the default) "glorot_uniform" init.

**Comment by carlthome** [comment]
Nice! Thanks for the links, @kylemcdonald. 

Actually, after thinking about it I figure training over several epochs is a much simpler way of optimizing the initial weights and that it probably wouldn't be worth the effort to do any more since in high-dimensional spaces getting stuck in local minima is improbable. http://www.kdnuggets.com/2015/11/theoretical-deep-learning.html

For other hyperparameters it seems very promising though. Again, thanks for the help!

**Comment by tboquet** [comment]
This package is also a good point to start: Hyperopt + Keras = [hyperas](https://github.com/maxpumperla/hyperas).

**Comment by carlthome** [comment]
@tboquet thanks! :smile:
